src/client.rs

```diff
@@ -256,8 +258,12 @@ impl ClientOram {
 
     pub fn setup_oram(&mut self, ct_size: usize) -> Result<Oram, Error> {
         // Computes the number of slots in the complete tree.
+        // We need ceil(nb_items / BUCKET_SIZE) buckets to store our elements.
+        // the next power of two minus one of this number is the size of the
+        // necessary complete tree.
         let slots_complete_tree =
-            ((1 << (self.nb_items.ilog2() + 1)) - 1) * BUCKET_SIZE;
+            ((1 << (udiv_ceil(self.nb_items, BUCKET_SIZE).ilog2() + 1)) - 1)
+                * BUCKET_SIZE;
```

Comment:
Same here, using the tree height or a new function `get_tree_height` would make these comments unnecessary:
```rust
let tree_slot_number = (1 << (get_tree_height(self.nb_items, BUCKET_SIZE) + 1) - 1) * BUCKET_SIZE;
```
Then the only thing that is to be explained is to the use of these `+1/-1`. But this may not need explanation with proper naming.

Reply:
What about atomic methods like
```rs
/// Performs ceiling division on usize type.
pub fn udiv_ceil(a: usize, b: usize) -> usize {
    (a as f32 / b as f32).ceil() as usize
}

/// The size of a complete tree is the next power of two minus one of the number
/// of items stored.
pub fn get_complete_tree_size(nb_items: usize, node_capacity: usize) -> usize {
    let nb_nodes_necessary = udiv_ceil(nb_items, node_capacity);

    // Getting next power of two strictly greater than `nb_nodes_necessary`.
    (1 << (nb_nodes_necessary.ilog2() + 1)) - 1
}

/// There are two times less leaves than there are nodes + 1 in a complete tree.
pub fn get_complete_tree_leaves(
    nb_items: usize,
    node_capacity: usize,
) -> usize {
    (get_complete_tree_size(nb_items, node_capacity) + 1) >> 1
}```
to skip overly complex lines with bit shifting +/-1 ?

Reply:
Please consider commit 69e63eb0a374594c30fa687a874c48e66f5a897e

Reply:
You did not update this line using your new functions.